src/openssl/ssl/context.cr

```diff
@@ -90,10 +90,21 @@ abstract class OpenSSL::SSL::Context
     #
     # For everything else this uses the defaults of your OpenSSL.
     # Use this only if undoing the defaults that `new` sets is too much hassle.
-    def self.insecure(method : LibSSL::SSLMethod = Context.default_method)
+    def self.insecure(method : LibSSL::SSLMethod = Context.default_method) : self
       super(method)
     end
 
+    # Parses a client context configuration from a hash-like interface.
+    #
+    # ```
+    # require "openssl"
+    #
+    # context = OpenSSL::SSL::Context::Client.parse({"key" => "private.key", "cert" => "certificate.crt", "ca" => "ca.pem"})
+    # ```
+    def self.parse(params) : self
```

Comment:
I would expect `T.parse` to be from `String -> T`.  I would suggest `T.from_hash` for this method.